docs/plugins/discovery-ec2.asciidoc

```diff
@@ -1,167 +1,215 @@
+be ignored by the discovery process.
 
-`availability_zones`::
+For example if you tag some EC2 instances with a tag called
+`my-elasticsearch-host` and set `host_type: tag:my-elasticsearch-host` then the
+`discovery-ec2` plugin will read each instance's host name from the
+`my-elasticsearch-host` tag.
+http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/Using_Tags.html[Read more
+about EC2 Tags].
 
-    Either a comma separated list or array based list of availability zones.
-    Only instances within the provided availability zones will be used in the
-    cluster discovery.
+--
 
-`any_group`::
+`discovery.ec2.availability_zones`::
 
-    If set to `false`, will require all security groups to be present for the
-    instance to be used for the discovery. Defaults to `true`.
+    A list of the names of the availability zones to use for discovery. Only
+    instances placed in one of the given availability zones will be used for
+    discovery.
 
-`node_cache_time`::
+[[discovery-ec2-filtering]]
+`discovery.ec2.tag.TAGNAME`::
+
++
+--
 
-    How long the list of hosts is cached to prevent further requests to the AWS API.
-    Defaults to `10s`.
+A list of the values of a tag called `TAGNAME` to use for discovery. If set,
+only instances that are tagged with one of the given values will be used for
+discovery. For instance, the following settings will only use nodes with a
+`role` tag set to `master` and an `environment` tag set to either `dev` or
+`staging`.
 
-*All* secure settings of this plugin are {ref}/secure-settings.html#reloadable-secure-settings[reloadable].
-After you reload the settings, an aws sdk client with the latest settings
-from the keystore will be used.
+[source,yaml]
+----
+discovery.ec2.tags.role: master
+discovery.ec2.tags.environment: dev,staging
+----
 
-[IMPORTANT]
-.Binding the network host
-==============================================
+One practical use for tag filtering is when an EC2 cluster contains many nodes
+that are not master-eligible {es} nodes. In this case, tagging the EC2 instances
+that _are_ running the master-eligible {es} nodes, and then filtering by that
+tag, will help discovery to run more efficiently.
 
-It's important to define `network.host` as by default it's bound to `localhost`.
+NOTE: The names of tags used for discovery may only contain ASCII letters,
+numbers, hyphens and underscores. In particular you cannot use tags whose name
+includes a colon.
 
-You can use {ref}/modules-network.html[core network host settings] or
-<<discovery-ec2-network-host,ec2 specific host settings>>:
+--
 
-==============================================
+`discovery.ec2.node_cache_time`::
 
-[[discovery-ec2-network-host]]
-===== EC2 Network Host
+    Sets the length of time for which the collection of discovered instances is
+    cached. Defaults to `10s`. This setting limits how frequently {es} requests
+    discovery information from the EC2 API. It is important to limit this
```

Comment:
This makes it sound like it is important that the user would explicitly configure limiting here. Instead I would make the phrasing more passive by saying something along the lines of "Limiting the request frequency is important because ..."

Reply:
Ok, adjusted in 88daba58001b46a154c335ba5093f9fac235b8c5.